modules/apps/frontend-js/frontend-js-importmaps-extender/src/main/java/com/liferay/frontend/js/importmaps/extender/internal/JsImportmapsContributorExtender.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.importmaps.extender.internal;
+
+import com.liferay.frontend.js.importmaps.extender.JsImportmapsContributor;
+import com.liferay.frontend.js.importmaps.extender.internal.servlet.taglib.JsImportmapsExtenderTopHeadDynamicInclude;
+import com.liferay.frontend.js.importmaps.extender.internal.servlet.taglib.JsImportmapsRegistration;
+import com.liferay.portal.kernel.json.JSONFactory;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.util.tracker.ServiceTracker;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(immediate = true, service = {})
+public class JsImportmapsContributorExtender {
```

Comment:
For now I'm only allowing to register importmaps programatically, but in the future we may want to let people register impormaps with a simple JSON file.

I'm not 100% sure, though, because we don't want the importmaps to become humongous, so we may prefer to make contributing to the importmaps less easy.

Reply:
> we may prefer to make contributing to the importmaps less easy.

Yeah I like the idea of not exposing it and making it easy yet. We can always add it in the future if it really is needed, but as we have seen with other config files of ours, it grows easily and harder to maintain across many teams/modules.

Reply:
Yep. We also don't want a huge importmap, because we used to have a similar endpoint for AMD loader v3 and we had to revamp it in AMD loader v4 to make it smaller. The problem is that if the importmap grows too much, the loading times of the home page can grow to inadmissible magnitudes.

Reply:
So, my idea is to restrict the importmap to deduplications and well-known APIs only.